src/ClientHandle.cpp

```diff
@@ -1120,34 +1116,30 @@ void cClientHandle::HandleLeftClick(int a_BlockX, int a_BlockY, int a_BlockZ, eB
 		When the user breaks a fire block, the client send the wrong block location.
 		We must find the right block with the face direction. */
 
-		int BlockX = a_BlockX;
-		int BlockY = a_BlockY;
-		int BlockZ = a_BlockZ;
-		AddFaceDirection(BlockX, BlockY, BlockZ, a_BlockFace);
+		Vector3i BlockPos = a_BlockPos;
+		AddFaceDirection(BlockPos, a_BlockFace);
 
-		if (cChunkDef::IsValidHeight(BlockY) && cBlockInfo::IsClickedThrough(m_Player->GetWorld()->GetBlock({ BlockX, BlockY, BlockZ })))
+		if (cChunkDef::IsValidHeight(BlockPos.y) && cBlockInfo::IsClickedThrough(m_Player->GetWorld()->GetBlock(BlockPos)))
 		{
-			a_BlockX = BlockX;
-			a_BlockY = BlockY;
-			a_BlockZ = BlockZ;
+			a_BlockPos = BlockPos;
 		}
 
 		if (
-			(Diff(m_Player->GetPosX(), static_cast<double>(a_BlockX)) > 6) ||
-			(Diff(m_Player->GetPosY(), static_cast<double>(a_BlockY)) > 6) ||
-			(Diff(m_Player->GetPosZ(), static_cast<double>(a_BlockZ)) > 6)
+			(Diff(m_Player->GetPosX(), static_cast<double>(a_BlockPos.x)) > 6) ||
+			(Diff(m_Player->GetPosY(), static_cast<double>(a_BlockPos.y)) > 6) ||
+			(Diff(m_Player->GetPosZ(), static_cast<double>(a_BlockPos.z)) > 6)
 		)
```

Comment:
Should this use one of our library functions rather than manual comparison to make things clearer (BoundingBox etc.)?